Shared/Samples/Analyze network with subnetwork trace/AnalyzeNetworkWithSubnetworkTraceView.Model.swift

```diff
@@ -0,0 +1,431 @@
+                } else if let value1 = value1 as? Int16,
+                          let value2 = value2 as? Int16 {
+                    return value1 == value2
+                } else {
+                    return false
+                }
+            }
+        }
+        
+        /// Adds a conditional expression.
+        func addConditionalExpression(
+            attribute: UtilityNetworkAttribute,
+            comparison: UtilityNetworkAttributeComparison.Operator,
+            value: Any
+        ) throws {
+            let convertedValue: Any
+            
+            if let codedValue = value as? CodedValue, attribute.domain is CodedValueDomain {
+                convertedValue = convertToDataType(value: codedValue.code!, dataType: attribute.dataType)
+            } else {
+                convertedValue = convertToDataType(value: value, dataType: attribute.dataType)
+            }
+            
+            if let expression = UtilityNetworkAttributeComparison(
+                networkAttribute: attribute,
+                operator: comparison,
+                value: convertedValue
+            ) {
+                traceConditionalExpressions.append(expression)
+                conditions.append(string(for: expression))
+            } else {
+                throw InitExpressionError()
+            }
+        }
+        
+        /// Deletes a conditional expression.
+        func deleteConditionalExpression(atOffsets indexSet: IndexSet) {
+            guard let index = indexSet.first else { return }
+            let condition = conditions[index]
+            conditions = conditions.filter { $0 != condition }
+            traceConditionalExpressions.remove(atOffsets: indexSet)
+        }
+        
+        /// Converts the values to matching data types.
+        /// - Note: The input value can either be an `CodedValue` populated from the left hand side
+        ///         attribute's domain, or a numeric value entered by the user.
+        /// - Parameters:
+        ///   - value: The right hand side value used in the conditional expression.
+        ///   - dataType: An `UtilityNetworkAttribute.DataType` enum case.
+        /// - Returns: Converted value.
+        func convertToDataType(value: Any, dataType: UtilityNetworkAttribute.DataType) -> Any {
+            switch dataType {
+            case .integer:
+                if let value = value as? Int16 {
+                    return value
+                } else if let value = value as? Int64 {
+                    return value
+                } else if let value = value as? Double {
+                    return Int64(value)
+                } else {
+                    return value as! Int64
+                }
+            case .float:
+                return value as! Float
+            case .double:
+                return value as! Double
+            case .boolean:
+                return value as! Bool
+            @unknown default:
```

Comment:
Thanks for approving this PR! I'd like to get #250 merged before releasing 200.2, so may need you to approve again shortly after. 🙂